statannot/comparisons_corrections/ComparisonsCorrection.py

```diff
@@ -0,0 +1,45 @@
+from statannot.comparisons_corrections.funcs import bonferroni, holm_bonferroni, benjamin_hochberg
+
+methods_names = {'bonferroni':        'Bonferroni',
+                 'bonf':              'Bonferroni',
+                 'Bonferroni':        'Bonferroni',
+                 'holm-bonferroni':   'Holm-Bonferroni',
+                 'HB':                'Holm-Bonferroni',
+                 'Holm-Bonferroni':   'Holm-Bonferroni',
+                 'benjamin-hochberg': 'Benjamin-Hochberg',
+                 'BH':                'Benjamin-Hochberg',
+                 'Benjamin-Hochberg': 'Benjamin-Hochberg'}
+
+# methods_data represents function and method type:
+# Type 0 means per-pvalue correction with modification
+# Type 1 means correction of interpretation of set of (ordered) pvalues
+
+methods_data = {'Bonferroni':        [bonferroni, 0],
+                'Holm-Bonferroni':   [holm_bonferroni, 1],
+                'Benjamin-Hochberg': [benjamin_hochberg, 1]}
+
+IMPLEMENTED_METHODS = [name for name in methods_names.keys()]
+
+
+def get_correction_method(name):
+    if name is None:
+        return None
+
+    if name not in IMPLEMENTED_METHODS:
+        raise NotImplementedError(f"Comparison {str(name)} not implemented.")
+
+    correction_method = methods_names[name]
+
+    return ComparisonsCorrection(correction_method,
+                                 *methods_data[correction_method])
+
+
+class ComparisonsCorrection:
+
+    def __init__(self, name, func, method_type):
+        self.name = name
+        self.func = func
+        self.type = method_type
+
+    def __call__(self, *args, **kwargs):
+        return self.func(*args, **kwargs)
```

Comment:
The idea is to keep things simple, as you noted. That is one task per function. 
The `bonferroni` function's job, for example, is to provide the modified values.
It could be used in any other place, context, even package if we want, as-is (with utils), and does not need a name or a type to work.
But as we need it, I wrote this, maybe too quickly ?
Another option would have been to use decorators, keeping things more simple.
We could certainly explore that !

The function documentation aspect is completely true, and to follow-up on your quick fix idea, unless I'm missing something, let me suggest:

```suggestion
class ComparisonsCorrection:

    def __init__(self, name, func, method_type):
        self.name = name
        self.func = func
        self.type = method_type
        self.__doc__ = f"ComparisonCorrection wrapper for the function " \
                       f"{func.__name__}, providing the attributes\n" \
                       f"    `name`: {self.name}\n" \
                       f"    `type`: {self.type} \n" \
                       f"With\n" \
                       f"    type 0 function: Per-pvalue correction with modification\n" \
                       f"    type 1 function: Correction of interpretation of set of (ordered) pvalues\n" \
                       f"------ Original function documentation ------ \n" \
                       f"{func.__doc__}"
               
    def __call__(self, *args, **kwargs):
        return self.func(*args, **kwargs)
```

Reply:
This is a good quick fix! You mentioned somewhere else on this PR that you were thinking of eventually factoring out the statistical calculations into their own submodule and handling what's essentially pretty-printing separately. I think this is a worthwhile goal.

After taking a second look at your `ComparisonsCorrection` class and the underlying `Comparisons` functions, I wonder if what we really have here is a polymorphic base class `ComparisonsCorrection` and a set of subclasses that should have a method `correct(p_values, num_comparisons='auto')`.

Something like this, for example:
```python
class HolmBonferroniCorrection(ComparisonsCorrection):
    name = 'Holm-Bonferroni'
    type = 1
    def __init__(self, alpha):
        self.alpha = alpha
    def correct(p_values, num_comparisons='auto'):
        return holm_bonferroni(p_values, num_comparisons, self.alpha)
```

That way, any clients that use a `ComparisonsCorrection` object can remain blissfully unaware of which one they're using, and metadata used for formatting is baked directly into the correction object.

What do you think? Is this too complicated?

Reply:
You are right, this is what I meant in the following part's [comment](https://github.com/webermarcolivier/statannot/pull/44#discussion_r398474933):
> To continue on the use of the `ComparisonsCorrection` class, this will come handy to be able to specify more parameters without passing through to `add_annotation` but with first initializing the correction function and then passing it to `add_annotation`.

But the user has to know (and choose) which correction to use.